src/main/scala/rover/rdo/ObjectState.scala

```diff
@@ -1,29 +1,26 @@
 package rover.rdo
-
-import votingapp.{PollChoice, Votes}
-
+	
 trait ObjectState {
 
 }
 
-class AtomicObjectState extends ObjectState {
+abstract class AtomicObjectState[A <: AtomicObjectState[A]] extends ObjectState {
+	// binding this to a self-type A
+	this: A =>
```

Comment:
Why?

Reply:
It says that `this` is of type `A`. This allows us to do `operation.apply(this)` at line 19. Otherwise, `this` is not of type `A`, instead it's an `AtomicObjectState` while `Op` takes in an `A <: AtomicObjectState`.

I had included the following link in the commit description, but GitHub doesn't advertise commit descriptions. This was the eventual solution: https://tpolecat.github.io/2015/04/29/f-bounds.html

Reply:
Pretty thorough link!